### PR TITLE
fix: Don't use row index for accessing chunk data

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/12577 (can also be reproduces by syncing only `azure_compute_skus`)

It was a bit hard to for me to understand the code, but it seems we use `NewTableFromRecords` to map an array of records with length of `batch_size` to a single Arrow table. This means the resulting table will have `batch_size` number of rows, with each column having `number_of_schema_columns` chunks, where each chunk of arrays is of `batch_size` length. Each array that is a part of the chunk of arrays has the underlying data from the source (think of a chunk of a way to manage a collection of arrays).

See
https://github.com/cloudquery/arrow/blob/b7ee35bf784acdf7af50f7776ce8b9dd8117a4c2/go/arrow/array/table.go#L177
https://github.com/cloudquery/arrow/blob/b7ee35bf784acdf7af50f7776ce8b9dd8117a4c2/go/arrow/array/table.go#L193

The current code increments the row index for each array in the chunk:
https://github.com/cloudquery/cloudquery/blob/5fa1af103a6e0e0b62d05f4d5d6f902c23a9e03e/plugins/destination/mssql/queries/values.go#L17

and for each value in the chunk data array:
https://github.com/cloudquery/cloudquery/blob/5fa1af103a6e0e0b62d05f4d5d6f902c23a9e03e/plugins/destination/mssql/queries/values.go#L18

then uses that value as an index to access the data in each chunk.

As mentioned before each column's chunk references `batch_size` number of arrays so we cannot use that length as a indicator on how to access each array's data.

**The fix is to use the `row` index as the index to the preallocated row array, and `0` as the index to access the underlying data of each array in the chunk. This works since sources don't save data in arrays longer than 1, so we know there's always a single value.**

> Technically we should be "unwrapping" each data array, but that can be done in a follow up (and maybe remove the usage of `NewTableFromRecords` as we could create the query rows directly from the records array).

### CODE REVIEW

File: plugins/destination/mssql/queries/values.go [hunk 1]
```diff
@@ -14,14 +14,13 @@ func GetRows(table arrow.Table) ([][]any, error) {
 	for n := 0; n < int(table.NumCols()); n++ {
 		col := table.Column(n)
 		row := 0
-		for _, chunk := range col.Data().Chunks() {
-			for i := 0; i < chunk.Len(); i++ {
-				rows[row][n], err = getColValue(chunk, row)
```
Comment (disq) [inline plugins/destination/mssql/queries/values.go:19]: This comment might not make sense, but why not `getColValue(chunk, i)` instead?

Comment (erezrokah) [reply plugins/destination/mssql/queries/values.go:19]: Yeah, overwriting the row might not make sense either (even if it only runs a single iteration every time)

Comment (erezrokah) [reply plugins/destination/mssql/queries/values.go:19]: Maybe `We only support records with arrays of length == 1, so we always extract the first value`

**Review by disq — APPROVED** [review]

File: plugins/destination/mssql/queries/values.go (hunk 1 above)
Comment (candiduslynx) [inline plugins/destination/mssql/queries/values.go:19]: this was a one-liner of 
```go
rows[row][n], err = getColValue(chunk, i)
```

**Pull request merged**